inference/core/registries/roboflow.py

```diff
@@ -75,6 +85,30 @@ def get_model(self, model_id: ModelID, api_key: str) -> Model:
         return self.registry_dict[model_type]
 
 
+@ttl_cache(ttl=MODELS_CACHE_AUTH_CACHE_TTL, maxsize=MODELS_CACHE_AUTH_CACHE_MAX_SIZE)
+def _check_if_api_key_has_access_to_model(
+    api_key: str,
+    model_id: str,
+) -> bool:
+    _, version_id = get_model_id_chunks(model_id=model_id)
+    try:
+        if version_id is not None:
+            get_roboflow_model_data(
+                api_key=api_key,
+                model_id=model_id,
+                endpoint_type=ModelEndpointType.ORT,
+                device_id=GLOBAL_DEVICE_ID,
+            ).get("ort")
```

Comment:
The other use-case of the old authorizer in https://github.com/roboflow/inference-internal/tree/main/deploy/authorizer is rate-limiting each inference call (unfortunately it cannot do anything when its Workflows like /workflows/run because there is no API key in the header of query parameters).

Should we be adding an endpointin api.roboflow.com to check this and deny an API key if they are over their quota?